Cython/Compiler/ExprNodes.py

```diff
@@ -759,7 +759,7 @@ def make_owned_memoryviewslice(self, code):
         # TODO ideally this would be shared with "make_owned_reference"
         if not self.result_in_temp():
             code.put_incref_memoryviewslice(self.result(), self.type,
-                                            have_gil=True)
+                                            have_gil=self.in_nogil_context)
```

Comment:
It apparently makes no difference (i.e. probably isn't used in the tests), but I think it needs to be negated.
```suggestion
                                            have_gil=not self.in_nogil_context)
```